### PR TITLE
[fix] fix compile

### PR DESCRIPTION
**Pull request by qicosmos**
## Why

fix compile error on gcc10.2 or lower version;

fix compile error on msvc;

## What is changing

add no-unused-value when gcc10.2 or lower;

use (std::min) instead of std::min to avoid conflict on windows;

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -63,6 +63,7 @@ else()
         -Wno-deprecated-register
         -Wno-mismatched-new-delete
         -Wno-deprecated-declarations
+        -Wno-unused-value
```
Comment (ChuanqiXu9) [inline CMakeLists.txt:66]: Generally it is bad to have unused values, what is the warning? Can we try to fix that?

File: async_simple/coro/Collect.h [hunk 2]
```diff
@@ -371,7 +371,7 @@ inline auto collectAllWindowedImpl(size_t maxConcurrency,
     }
     size_t start = 0;
     while (start < input_size) {
-        size_t end = std::min(input_size, start + maxConcurrency);
+        size_t end = (std::min)(input_size, start + maxConcurrency);
```
Comment (ChuanqiXu9) [inline async_simple/coro/Collect.h:374]: Let's add a comment to explain this.

File: CMakeLists.txt (hunk 1 above)
Comment (qicosmos) [reply CMakeLists.txt:66]:
```
[build] [ 85%] Building CXX object demo_example/CMakeFiles/CountChar.dir/CountChar.cpp.o
[build] async_simple/demo_example/CountChar.cpp: In function ‘async_simple::coro::Lazy<int> CountTextChar(const Texts&, char)’:
[build] async_simple/demo_example/CountChar.cpp:50:1: error: statement has no effect [-Werror=unused-value]
[build]    50 | }
[build]       | ^
[build] At global scope:
```

```c++
Lazy<int> CountTextChar(const Texts &Content, char c) {
    int Result = 0;
    for (const auto &line : Content)
        Result += co_await CountLineChar(line, c);
    co_return Result;
}
```

I think it's a bug of gcc10.

Comment (ChuanqiXu9) [reply CMakeLists.txt:66]: Agreed. But disable `-Wno-unused-value` looks over acted. I feel we can add a `(void)Result;` before `co_return` to eliminate the warning.

**Pull request merged**